v2/common/runner.go

```diff
@@ -23,9 +23,15 @@ import (
 // depending on request path.
 type ClientWrapper struct {
 	*http.Client
+	Dev bool
 }
 
 func (c *ClientWrapper) Do(r *http.Request) (*http.Response, error) {
+	// when in dev mode, add the X-Rescale-Environment header.
+	if c.Dev {
+		r.Header.Set("X-Rescale-Environment", "dev")
```

Comment:
It's a little jankier, but I think it may be better to add an `X_RESCALE_ENVIRONMENT` env var that you check, and document that in the README, rather than expose it has an option. My thinking here is that very few users will want to set this and so it probably just confuses them.

And in general, fewer options is always best.

```suggestion
	if rescaleEnv := os.Getenv("X_RESCALE_ENVIRONMENT"); rescaleEnv != "" {
		r.Header.Set("X-Rescale-Environment", rescaleEnv)
```

Reply:
I agree that a dedicated flag seems like overkill (and also exposes some internal system details to end users outside Rescale that is going to be confusing), undocumented env var passing seems sufficient

(I was going to suggest an undocumented config context var but that sounds like more work than what Hunter has above)

Reply:
I like this idea more. Updated 👍

Reply:
Any thoughts on using `LookupEnv` instead of `Getenv`? Probably not the biggest deal, but the slight difference is `LookupEnv` would distinguish of a user set the value explicitly to the  "" or if it wasn't set at all. Might be able to throw an error if it is explicitly set to "".

Reply:
(as discussed in Slack)

my two cents is that this is one of those cases where we benefit from following Go's style of zero valued types instead of null. I.e. here, we don't actually need/want to distinguish between the env var being empty string and the env var not being set at all - in either case, we'd skip adding the header.